common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -85,11 +63,15 @@
     }
   }
 
+  private static final int majorVersion;
+  static {
+    // Split java.version on non-digit chars:
+    majorVersion = Integer.parseInt(System.getProperty("java.version").split("\\D+")[0]);
+  }
```

Comment:
For reusing, this is moved out from the below.

Reply:
You don't need a static init block here, but it's just a nit. Just assign the variable.

Reply:
Thanks. Yep. I fixed it.